engine/src/ui/canvas/plugin.cpp

```diff
@@ -116,4 +138,52 @@ void cubos::engine::uiCanvasPlugin(Cubos& cubos)
                 window->renderDevice().clearDepth(1);
             }
         });
+
+    cubos.system("draw ui").tagged(uiEndTag).with<UIElement>().related<ChildOf>().with<UICanvas>().call(
+        [](Query<UIElement&, const UICanvas&> query, Query<Entity, UICanvas&> canvasQuery,
+           const core::io::Window& window) {
+            for (auto [entity, canvas] : canvasQuery)
+            {
+                std::map<DrawType*, std::vector<DrawCommand>> commands;
+
+                for (auto [element, _] : query.pin(1, entity))
+                {
+                    for (auto entry : element.drawList)
+                    {
+                        commands[&entry.type].push_back(entry.command);
+                    }
+                }
+
+                window->renderDevice().setViewport(0, 0, static_cast<int>(window->framebufferSize()[0]),
+                                                   static_cast<int>(window->framebufferSize()[1]));
```

Comment:
Here you should access the render target's size instead of the window's. Just add the `RenderTarget` component to the canvasQuery, as they're part of the same entity.